cmake/AddBipedalLocomotionControllersComponent.cmake

```diff
@@ -0,0 +1,104 @@
+# Copyright (C) 2019 Istituto Italiano di Tecnologia (IIT). All rights reserved.
+# This software may be modified and distributed under the terms of the
+# GNU Lesser General Public License v2.1 or any later version.
+
+function(add_bipedal_component)
+
+  set(options IS_INTERFACE)
+  set(oneValueArgs NAME)
+  set(multiValueArgs
+    SOURCES
+    PUBLIC_HEADERS
+    PRIVATE_HEADERS
+    PUBLIC_LINK_LIBRARIES
+    PRIVATE_LINK_LIBRARIES
+    SUBDIRECTORIES)
+
+  set(prefix "bipedal_component")
+
+  cmake_parse_arguments(${prefix}
+    "${options}"
+    "${oneValueArgs}"
+    "${multiValueArgs}"
+    ${ARGN})
+
+  set(name ${${prefix}_NAME})
+  set(is_interface ${${prefix}_IS_INTERFACE})
+  set(sources ${${prefix}_SOURCES})
+  set(public_headers ${${prefix}_PUBLIC_HEADERS})
+  set(private_headers ${${prefix}_PRIVATE_HEADERS})
+  set(public_link_libraries ${${prefix}_PUBLIC_LINK_LIBRARIES})
+  set(private_link_libraries ${${prefix}_PRIVATE_LINK_LIBRARIES})
+  set(subdirectories ${${prefix}_SUBDIRECTORIES})
+
+  # The building process is different if the component is an INTERFACE
+  if(is_interface)
```

Comment:
Note that a lot of  this logic can be simplified once we can depend on CMake 3.15, in which the PUBLIC_HEADER options are available also for INTERFACE  libraries: https://gitlab.kitware.com/cmake/cmake/-/merge_requests/3181 .

Reply:
What do you think? Shall we add  CMake 3.15 as a dependency?  Is it too strong?

Reply:
No, I just wanted to put a remaineder that when we require a newer CMake, we can simplify this part.